js/indicators/tema.src.js

```diff
@@ -0,0 +1,235 @@
+'use strict';
+import H from '../parts/Globals.js';
+import '../parts/Utilities.js';
+
+var isArray = H.isArray,
+    EMA = H.seriesTypes.ema,
+    correctFloat = H.correctFloat;
+
+/**
+* The TEMA series Type
+*
+* @constructor seriesTypes.tema
+* @augments seriesTypes.ema
+*/
+
+H.seriesType('tema', 'ema',
+    /**
+     * Normalized average true range indicator (NATR). This series requires
+     * `linkedTo` option to be set.
+     *
+     * Requires https://code.highcharts.com/stock/indicators/ema.js.
+     *
+     * @extends plotOptions.ema
+     * @product highstock
+     * @sample {highstock} stock/indicators/tema TEMA indicator
+     * @excluding
+     *      allAreas,colorAxis,compare,compareBase,joinBy,keys,stacking,
+     *      showInNavigator,navigatorOptions,pointInterval,
+     *      pointIntervalUnit,pointPlacement,pointRange,pointStart,joinBy
+     * @since 7.0.0
+     * @optionparent plotOptions.tema
+     */
+    {}, {
+        getValues: function (series, params) {
+            var period = params.period,
+                doubledPeriod = 2 * period,
+                tripledPeriod = 3 * period,
+                xVal = series.xData,
+                yVal = series.yData,
+                yValLen = yVal ? yVal.length : 0,
+                EMApercent = (2 / (period + 1)),
+                index = -1,
+                sum = 0,
+                ema = 0,
+                SMA = 0,
+                EmaArr = [],
```

Comment:
- Use consistent names of variables i.e EMA... instead of Ema
- EmaArr is not clear, better will be EMApoints / EMAvalues.